app/actors/hyrax/actors/apply_permission_template_actor.rb

```diff
@@ -13,8 +13,23 @@ def create(env)
       private
 
         def add_edit_users(env)
-          return if env.attributes[:source_id].blank?
-          template = Hyrax::PermissionTemplate.find_by!(source_id: env.attributes[:source_id])
+          add_admin_set_participants(env)
+          add_collection_participants(env)
+        end
+
+        def add_admin_set_participants(env)
+          return if env.attributes[:admin_set_id].blank?
+          template = Hyrax::PermissionTemplate.find_by!(source_id: env.attributes[:admin_set_id])
+          env.curation_concern.edit_users += template.agent_ids_for(agent_type: 'user', access: 'manage')
+          env.curation_concern.edit_groups += template.agent_ids_for(agent_type: 'group', access: 'manage')
+          env.curation_concern.read_users += template.agent_ids_for(agent_type: 'user', access: 'view')
+          env.curation_concern.read_groups += template.agent_ids_for(agent_type: 'group', access: 'view')
+        end
+
+        def add_collection_participants(env)
+          return if env.attributes[:collection_id].blank?
+          collection_id = env.attributes.delete(:collection_id)
+          template = Hyrax::PermissionTemplate.find_by!(source_id: collection_id)
```

Comment:
Extract a method that handles copying template's agend_ids to the curation_concern

```ruby
def add_admin_set_participants(env)
  return if env.attributes[:admin_set_id].blank?
  template = Hyrax::PermissionTemplate.find_by!(source_id: env.attributes[:admin_set_id])
  add_template_participants(curation_concern: env.curation_concern, template: template)
end

def add_collection_participants(env)
  return if env.attributes[:collection_id].blank?
  collection_id = env.attributes.delete(:collection_id)
  template = Hyrax::PermissionTemplate.find_by!(source_id: collection_id)
  add_template_participants(curation_concern: env.curation_concern, template: template)
end

def add_template_participants(curation_concern:, template:)
  curation_concern.edit_users += template.agent_ids_for(agent_type: 'user', access: 'manage')
  curation_concern.edit_groups += template.agent_ids_for(agent_type: 'group', access: 'manage')
  curation_concern.read_users += template.agent_ids_for(agent_type: 'user', access: 'view')
  curation_concern.read_groups += template.agent_ids_for(agent_type: 'group', access: 'view')
end
```

Reply:
✅